src/LibraryUtilities.ts

```diff
@@ -566,48 +285,244 @@ export function updateSections(oldLayoutSpecs: any, newLayoutSpecs: any): void {
+
+    let results: ItemData[] = [];
+
+    for (let layoutElement of layoutElements) {
+        let layoutData = new ItemData(layoutElement.text);
+        layoutData.constructFromLayoutElement(layoutElement);
+
+        // Sections that don't show its header will be expanded by default.
+        if (layoutData.itemType === "section" && !layoutData.showHeader) {
+            layoutData.expanded = true;
+        }
+
+        if (parentItem) {
+            parentItem.appendChild(layoutData);
+        }
+
+        if (layoutElement.include.length > 0) {
+            for (let include of layoutElement.include) {
+                let pair = new IncludeItemPair();
+                pair.include = include;
+                pair.parentItem = layoutData;
+                pairs.push(pair);
+            }
+        }
+
+        if (layoutElement.childElements.length > 0) {
+            convertLayoutElementToItemData(layoutElement.childElements, pairs, layoutData);
+        }
+
+        results.push(layoutData);
+    }
+
+    return results;
+}
+
+/**
+ * This method will merge all typeListNodes under corresponding library items based on includeItemPairs.
+ * Each includeItemPair stores includeInfo and the parent item of this includeInfo.
+ * 
+ * When iterating through includeItemPairs, comparison will happen between fullyQualifiedName of typeListNode 
+ * and path of the include info of each pair.. 
+ * 
+ * If there is a match of them, which means they are exactly the same, this type node will be merged to 
+ * the parent item of the includeInfo. 
+ * 
+ * If there is not a match, but fullyQualifiedName includes path, this type node will
+ * first be built from name, and then merged to the parent item.
+ * For example, if fullayQualifiedName is A.B.C.D, path is A.B, the item to be merged would have the 
+ * following structure.
+ * 
+ * - parentItem
+ *  |- B
  *      |- C
- *      |- D
+ *          |- D
  * 
- * @param {TypeListNode[]} allNodes 
- * All the typeListNodes
+ * If there is not a match or include, skip to next includeInfo/typeListNode, depending on their 
+ * lexicographical order.
  * 
- * @param {LayoutElement} section
- * The section that the converted items will be under
+ * @param {TypeListNode[]} typeListNodes 
+ * An array of TypelistNode sorted in alphabetical order
  * 
- * @returns
- * Returns a single ItemData that contains the new nodes in its childItems.
+ * @param {IncludeItemPairs[]} includeItemPairs
+ * An array of IncludeItemPair sorted in alphabetical order of include.path
  */
-export function convertToMiscSection(allNodes: TypeListNode[], section: LayoutElement): ItemData {
+export function constructFromIncludeInfo(typeListNodes: TypeListNode[], includeItemPairs: IncludeItemPair[]) {
+    let t = 0;
+    let prefix = "://";
+    let nodeMatch = false;
+    let nodeInculde = false;
+
+    // previousIncludeParts is the includeInfo path parts of the pair before current pair.
+    let previousIncludeParts: string[] = [];
```

Comment:
I could not follow this function, I will let this in, and debug it to see what it does, and then add necessary comments.